frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileReader.java

```diff
@@ -0,0 +1,34 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.intellij.openapi.vfs.VirtualFileManager;
+import io.vavr.collection.List;
+import lombok.CustomLog;
+import lombok.val;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutReader;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@CustomLog
+public class MacheteFileReader {
+  @UIThreadUnsafe
```

Comment:
TBH I'd say it's safe for UI thread. Since we are even expected to **write** to VFS on UI thread (via `WriteAction`s)... then reading is even safer. What are the UI-thread-unsafe calls reported by ArchUnit in this method?

Reply:
ArchUnit reports all the operations connected with streams (package `java.io`) as unsafe. I deleted the `UIThreadUnsafe` annotations and put `IgnoreUIThreadUnsafeCalls` where it was needed.

But maybe the ArchUnit test should be somehow refactored based on this new idea that those actions are not unsafe and even need to be executed on UIThread?

Reply:
> But maybe the ArchUnit test should be somehow refactored based on this new idea that those actions are not unsafe and even need to be executed on UIThread?

Hmmm theoretically it's just a matter of updating `com.virtuslab.archunit.UIThreadUnsafeMethodInvocationsTestSuite#whitelistedMethodFullNames_java_io`... could you list the specific method from `java.io` reported here? so that we can decide whether:
1. keep `@IgnoreUIThreadUnsafeCalls` here in `frontend:file` method
or rather
2. update `whitelistedMethodFullNames_java_io`

Reply:
From `MacheteFileWriter`: `{"java.io.BufferedOutputStream.<init>(java.io.OutputStream)", "java.io.BufferedOutputStream.close()"}`

From `MacheteFileReader`: `"java.io.InputStream.close()"`

No stream constructors reported since we use `VirtualFile.getInputStream()` (not from java.io).

Reply:
Okay, I think all of `java.io.BufferedOutputStream.<init>(java.io.OutputStream)`, `java.io.BufferedOutputStream.close()` and `java.io.InputStream.close()` can be safely added to `whitelistedMethodFullNames_java_io`, thus removing the need for `@IgnoreUIThreadUnsafeCalls`